src/util/bip32.rs

```diff
@@ -167,6 +167,21 @@ impl fmt::Display for ChildNumber {
     }
 }
 
+impl FromStr for ChildNumber {
+    type Err = Error;
+
+    fn from_str(inp: &str) -> Result<ChildNumber, Error> {
+        match inp.chars().last().map_or(false, |l| l == '\'' || l == 'h') {
+            true => Ok(ChildNumber::from_hardened_idx(
```

Comment:
Perhaps we can put the `Ok()` outside of the match so we don't have to repeat ourselves in both cases.